frontend/src/pages/functional_components/NewExperimentFC.tsx

```diff
@@ -97,17 +94,11 @@ export function NewExperimentFC(props: NewExperimentFCProps) {
         open: true,
       });
     }
+    // Only trigger this effect when search string parameters change.
+    // Do not rerun this effect if updateSnackbar callback has changes to avoid re-rendering.
+    // Do not rerun this effect if pipelineId has changes to avoid re-rendering.
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [experimentResponse]);
-
-  // Handle the error when createExperiment() is failed
-  useEffect(() => {
-    if (!experimentName) {
-      setErrorMessage('Experiment name is required');
-    } else {
-      setErrorMessage('');
-    }
-  }, [experimentName]);
+  }, [experimentResponse, latestVersion]);
```

Comment:
I think once the `pipelineId` is changed, the `latestVersion` will change too (cause re-render as well). Not sure if it's good to include `latestVersion` in the dependency array.

Reply:
`pipelineId` is not the only reason that causes `latestVersion` to change. `latestVersion` is being pulled from `useQuery`, and the query can be executed any time and multiple times during the page is visible. Since we are using  `latestVersion` in this `useEffect` code block, I added the `latestVersion` to the dependency array.